### PR TITLE
b1

### PR DESCRIPTION
**Pull request by Oleksandr-Radovets**

### CODE REVIEW

File: src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Comment (Rommelua) [inline src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java:17]: add implementation

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookwebstoryspringboot/entity/Book.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    private String description;
+    private String coverImage;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
```
Comment (sokimaaa) [inline src/main/java/com/example/bookwebstoryspringboot/entity/Book.java:33]: I would suggest using the Lombok library to reduce boilerplate code, e.g. 
you can replace all getters and setters with only two annotations from Lombok - @Getter and @Setter

File: src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return book;
+        } catch (Exception e) {
+            if (entityTransaction != null) {
+                entityTransaction.rollback();
+            } else {
+                throw new RuntimeException("Can't save " + book + " " + e);
+            }
+        }
+        return book;
+    }
```
Comment (sokimaaa) [inline src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java:34]:
```suggestion
    public Book save(Book book) {
        EntityTransaction entityTransaction = null;

        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
            entityTransaction = entityManager.getTransaction();
            entityManager.persist(book);
            entityTransaction.commit();
        } catch (Exception e) {
            if (entityTransaction != null) {
                entityTransaction.rollback();
            } else {
                throw new RuntimeException("Can't save " + book + " " + e);
            }
        }
        return book;
    }
```

File: src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            TypedQuery<Book> query = entityManager.createQuery("SELECT b.id, b.description,"
+                    + "b.title,b.author,"
+                    + "b.coverImage, b.isbn, b.price FROM Book b", Book.class);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("there are no books");
+
+        }
+
```
Comment (sokimaaa) [inline src/main/java/com/example/bookwebstoryspringboot/repository/impl/BookRepositoryImpl.java:47]: remove redundant empty line
```suggestion
        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
            TypedQuery<Book> query = entityManager.createQuery("SELECT b.id, b.description,"
                    + "b.title,b.author,"
                    + "b.coverImage, b.isbn, b.price FROM Book b", Book.class);
            return query.getResultList();
        } catch (Exception e) {
            throw new RuntimeException("there are no books");
        }
```

File: src/main/java/com/example/bookwebstoryspringboot/entity/Book.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import java.math.BigDecimal;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+public class Book {
+
+    @Id
```
Comment (Rommelua) [inline src/main/java/com/example/bookwebstoryspringboot/entity/Book.java:17]:
```suggestion
public class Book {
    @Id
```

**Review by Rommelua — APPROVED** [review]

**Pull request merged**